src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,1070 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Do not use this. Use [bit-fields](https://docs.rs/bit-fields).
+#![warn(clippy::pedantic, clippy::restriction)]
+#![allow(
+    clippy::non_ascii_literal,
+    clippy::blanket_clippy_restriction_lints,
+    clippy::implicit_return,
+    clippy::items_after_statements
+)]
+use std::collections::HashSet;
+use std::convert::TryFrom;
+use std::fmt::{self, Write};
+
+use proc_macro2::{Delimiter, Group, TokenStream, TokenTree};
+use proc_macro_error::Level;
+use quote::quote;
+
+/// Type used to ensure safety by avoiding some arithmetic operations.
+///
+/// Without this type we would parse and store the size of the bitfield as a `u8`. The following
+/// operations would then use the generic `u8` implementations which the compiler cannot guarantee
+/// are safe. By using this enum to restrict the type we can avoid these operations. This is a
+/// small point, but helps a little.
+#[derive(Debug, Clone, Copy)]
+enum DataType {
+    /// `u8`
+    U8 = 8,
+    /// `u16`
+    U16 = 16,
+    /// `u32`
+    U32 = 32,
+    /// `u64`
+    U64 = 64,
+    /// `u128`
+    U128 = 128,
+}
+impl DataType {
```

Comment:
Could we maybe place DataType into a separate file called helpers or smth similar? The same for MultilineString

Reply:
Along with some other big changes, I've split the crate into 3 file, `builder.rs`, `utils.rs` and `lib.rs`. Moving `DataType` and `MultilineString` to `utils.rs`.